File: code/ruby/erlectricity/echo.erl
-module(echo).
-export([test/1]).

test(W) ->
  Cmd = "ruby echo.rb",
  Port = open_port({spawn, Cmd}, [{packet, 4}, nouse_stdio, exit_status, binary]),
  Payload = term_to_binary({echo, W }), %%<<"hello world!">>}),
  port_command(Port, Payload),
  receive
    {Port, {data, Data}} ->
      {result, Text} = binary_to_term(Data),
      io:format("~p~n", [Text])
  end.

File: code/erlang/erlectricity/echo.rb
require 'rubygems'
require 'erlectricity'

receive do |f|
  f.when([:echo, String]) do |text|
    f.send!([:result, "You said text: #{text}"])
    f.receive_loop
  end

  f.when([:echo, Fixnum]) do |num|
    f.send!([:result, "You said num: #{num}"])
    f.receive_loop
  end

  f.when([:echo, Array]) do |num|
    f.send!([:result, "You said num: #{num}"])
    f.receive_loop
  end
end
